core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java

```diff
@@ -147,4 +149,32 @@ public void testKeysWithNulls() {
 
     assertThat(map.remove(record3)).isEqualTo("aaa");
   }
+
+  @Test
+  public void testEqualsAndHashCode() {
```

Comment:
Can we do a quick addition of a test for empty map equality? I am pretty sure that is fine but just to be complete

Reply:
Sure, I'll add that.